tests/fixtures/page-router/pages/api/revalidate.js

```diff
@@ -1,6 +1,9 @@
 export default async function handler(req, res) {
   try {
-    await res.revalidate('/static/revalidate-manual')
+    // res.revalidate returns a promise that can be awaited to wait for the revalidation to complete
+    // if user doesn't await it, we still want to ensure the revalidation is completed, so we internally track
+    // this as "background work" to ensure it completes before function suspends execution
+    res.revalidate('/static/revalidate-manual')
```

Comment:
I think it makes sense to have 2 cases here so in case those e2e tests fail it's easier to tell wether there is overall problem with revalidation or specifically with not awaited revalidation. Adding variants here would be much smoother after https://github.com/netlify/next-runtime/pull/2495 is in as it does add ~`test.each` kind of setup to on-demand revalidation tests where it's super easy to add another case, so I'll get this one in first and then revert this change and instead add another API endpoint that doesn't await

Reply:
https://github.com/netlify/next-runtime/pull/2490/commits/dd43ce8491f6642cdbb09d4a0d4bccaf29235ef5 added separate case for not awaiting `res.revalidate` on top of existing ones